packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -80,3 +93,608 @@ class CupertinoColors {
+      separator: CupertinoDynamicColor.resolve(separator, context),
+      opaqueSeparator: CupertinoDynamicColor.resolve(opaqueSeparator, context),
+      link: CupertinoDynamicColor.resolve(link, context),
+      systemBlue: CupertinoDynamicColor.resolve(systemBlue, context),
+      systemGreen: CupertinoDynamicColor.resolve(systemGreen, context),
+      systemIndigo: CupertinoDynamicColor.resolve(systemIndigo, context),
+      systemOrange: CupertinoDynamicColor.resolve(systemOrange, context),
+      systemPink: CupertinoDynamicColor.resolve(systemPink, context),
+      systemPurple: CupertinoDynamicColor.resolve(systemPurple, context),
+      systemRed: CupertinoDynamicColor.resolve(systemRed, context),
+      systemTeal: CupertinoDynamicColor.resolve(systemTeal, context),
+      systemYellow: CupertinoDynamicColor.resolve(systemYellow, context),
+      systemGray: CupertinoDynamicColor.resolve(systemGray, context),
+      systemGray2: CupertinoDynamicColor.resolve(systemGray2, context),
+      systemGray3: CupertinoDynamicColor.resolve(systemGray3, context),
+      systemGray4: CupertinoDynamicColor.resolve(systemGray4, context),
+      systemGray5: CupertinoDynamicColor.resolve(systemGray5, context),
+      systemGray6: CupertinoDynamicColor.resolve(systemGray6, context),
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    return other.runtimeType == runtimeType
+        && label == label
+        && secondaryLabel == secondaryLabel
+        && tertiaryLabel == tertiaryLabel
+        && quaternaryLabel == quaternaryLabel
+        && systemFill == systemFill
+        && secondarySystemFill == secondarySystemFill
+        && tertiarySystemFill == tertiarySystemFill
+        && quaternarySystemFill == quaternarySystemFill
+        && placeholderText == placeholderText
+        && systemBackground == systemBackground
+        && secondarySystemBackground == secondarySystemBackground
+        && tertiarySystemBackground == tertiarySystemBackground
+        && systemGroupedBackground == systemGroupedBackground
+        && secondarySystemGroupedBackground == secondarySystemGroupedBackground
+        && tertiarySystemGroupedBackground == tertiarySystemGroupedBackground
+        && separator == separator
+        && opaqueSeparator== opaqueSeparator
+        && link == link
+        && systemBlue == systemBlue
+        && systemGreen == systemGreen
+        && systemIndigo == systemIndigo
+        && systemOrange == systemOrange
+        && systemPink == systemPink
+        && systemPurple == systemPurple
+        && systemRed == systemRed
+        && systemTeal == systemTeal
+        && systemYellow == systemYellow
+        && systemGray == systemGray
+        && systemGray2 == systemGray2
+        && systemGray3 == systemGray3
+        && systemGray4 == systemGray4
+        && systemGray5 == systemGray5
+        && systemGray6 == systemGray6;
```

Comment:
I wish there was a way in Dart to avoid huge lists of params like this.  It seems easy to forget to update these if a new parameter gets added, for example.

Reply:
https://github.com/dart-lang/language/issues/314 this? Would be really nice to have synthesized default implementations for stuff like this

Reply:
Exactly, I guess I'm not the first person to think of that haha.